src/Azure/Orleans.Clustering.CosmosDB/AzureCosmosDBGatewayListProvider.cs

```diff
@@ -0,0 +1,79 @@
+using System.Net;
+using Orleans.Messaging;
+using Orleans.Clustering.CosmosDB.Models;
+
+namespace Orleans.Clustering.CosmosDB;
+
+internal class AzureCosmosDBGatewayListProvider : IGatewayListProvider
+{
+    private readonly ILogger _logger;
```

Comment:
Wondering if we could potentially introduce some small abstraction layers here to allow this class to have some unit tests covering it as well as the functional tests? These abstraction layer could be added to the `Shared` project and used then across the different providers.

Reply:
Agreed another PR for this makes sense.